service/gcs/stdio/stdio.go

```diff
@@ -0,0 +1,155 @@
+package stdio
+
+import (
+	"io"
+	"os"
+	"sync"
+
+	"github.com/Microsoft/opengcs/service/gcs/transport"
+	"github.com/Sirupsen/logrus"
+	"github.com/pkg/errors"
+)
+
+// ConnectionSet is a structure defining the readers and writers the Core
+// implementation should forward a process's stdio through.
+type ConnectionSet struct {
+	In, Out, Err transport.Connection
+}
+
+// Close closes each stdio connection.
+func (s *ConnectionSet) Close() error {
+	var err error
+	if s.In != nil {
+		if cerr := s.In.Close(); cerr != nil && err == nil {
+			err = errors.Wrap(cerr, "failed Close on stdin")
+		}
+		s.In = nil
+	}
+	if s.Out != nil {
+		if cerr := s.Out.Close(); cerr != nil && err == nil {
+			err = errors.Wrap(cerr, "failed Close on stdout")
+		}
+		s.Out = nil
+	}
+	if s.Err != nil {
+		if cerr := s.Err.Close(); cerr != nil && err == nil {
+			err = errors.Wrap(cerr, "failed Close on stderr")
+		}
+		s.Err = nil
+	}
+	return err
+}
+
+// FileSet contains os.File fields for stdio.
+type FileSet struct {
+	In, Out, Err *os.File
+}
+
+// Close closes all the FIleSet handles.
+func (fs *FileSet) Close() error {
+	var err error
+	if fs.In != nil {
+		if cerr := fs.In.Close(); cerr != nil && err == nil {
+			err = errors.Wrap(cerr, "failed Close on stdin")
+		}
+		fs.In = nil
+	}
+	if fs.Out != nil {
+		if cerr := fs.Out.Close(); cerr != nil && err == nil {
+			err = errors.Wrap(cerr, "failed Close on stdout")
+		}
+		fs.Out = nil
+	}
+	if fs.Err != nil {
+		if cerr := fs.Err.Close(); cerr != nil && err == nil {
+			err = errors.Wrap(cerr, "failed Close on stderr")
+		}
+		fs.Err = nil
+	}
+	return err
+}
+
+// Files returns a FileSet with an os.File for each connection
+// in the connection set.
+func (s *ConnectionSet) Files() (_ *FileSet, err error) {
```

Comment:
I'm not sure how common the pattern used in the return type here is, but it would be clearer at least to me for the return type to be (*FileSet, error), and just have a "var err error" line at the beginning of the function. From what I've read, named return values are mostly intended for documentation rather than functionality.